plugins/global-search/src/components/SearchScene.tsx

```diff
@@ -59,28 +53,33 @@ export function SearchScene() {
     useEffect(() => {
         void framer.showUI(
             getPluginUiOptions({
-                query: queryToUse,
-                hasResults,
-                withMessage: noResultsState !== false,
+                query: deferredQuery,
+                hasResults: hasResults,
+                withMessage: showNoResults,
             })
         )
-    }, [queryToUse, hasResults, noResultsState])
+    }, [deferredQuery, hasResults, showNoResults])
 
     return (
         <main className="flex flex-col h-full">
             <SelectionProvider>
                 <div
                     className={cn(
                         "flex gap-2 border-divider-light dark:border-divider-dark border-y py-3 mx-3 transition-colors items-center",
-                        !queryToUse && "border-b-transparent dark:border-b-transparent"
+                        !deferredQuery && "border-b-transparent dark:border-b-transparent"
                     )}
                 >
                     <SearchInput value={query} onChange={handleQueryChange} />
 
                     <div
-                        title="Indexing..."
-                        className="aria-hidden:opacity-0 transition flex items-center justify-center"
-                        aria-hidden={!isIndexingWithMinimumDuration}
+                        title="Searching…"
+                        className={cn(
+                            "aria-hidden:opacity-0 transition-opacity flex items-center justify-center",
+                            !isInitialIndexing && "delay-500"
```

Comment:
I'm undecided if this delay (based on "how long is the filterer running" vs "how long since the last character".